BaseStation/MainWindow/Settings/SettingsFile.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Serialization;
+
+namespace HuskyRobotics.UI
+{
+    /// <summary>
+    /// Settings along with a file and auto-save functionality.
+    /// </summary>
+    public class SettingsFile
+    {
+        public string FileName { get; private set; }
+        public Settings Settings { get; private set; }
+
+        public SettingsFile(string fileName)
+        {
+            FileName = fileName;
+
+            if (File.Exists(FileName))
+            {
+                try
+                {
+                    XmlSerializer serializer = new XmlSerializer(typeof(Settings));
```

Comment:
Do we want to serialize in XML? XML is kinda ugly and space-inefficient. Maybe we can get away with binary serialization?